File: FireflyDevice/FireflyDevice/FDBundleManager.m
//
//  FDBundleManager.m
//  FireflyDevice
//
//  Created by Denis Bohm on 2/14/14.
//  Copyright (c) 2014 Firefly Design LLC / Denis Bohm. All rights reserved.
//

#import <FireflyDevice/FDBundleManager.h>

static NSMutableArray *libraryBundles = nil;

@implementation FDBundleManager

+ (void)addLibraryBundle:(id<FDBundleInfo>)bundle
{
    if (libraryBundles == nil) {
        libraryBundles = [NSMutableArray array];
    }
    [libraryBundles addObject:bundle];
}

+ (NSArray *)allLibraryBundles
{
    return [NSArray arrayWithArray:libraryBundles];
}

@end

File: Windows/FireflyDevice/FDBundleManager.cpp
//
//  FDBundleManager.cpp
//  FireflyDevice
//
//  Created by Denis Bohm on 2/14/14.
//  Copyright (c) 2014 Firefly Design LLC / Denis Bohm. All rights reserved.
//

#include "FDBundleManager.h"

namespace FireflyDesign {

	static std::vector<std::shared_ptr<FDBundleInfo>> libraryBundles;

	void FDBundleManager::addLibraryBundle(std::shared_ptr<FDBundleInfo> bundle)
	{
		libraryBundles.push_back(bundle);
	}

	std::vector<std::shared_ptr<FDBundleInfo>> allLibraryBundles()
	{
		return libraryBundles;
	}

}